threadpoolctl.py

```diff
@@ -196,19 +196,25 @@ def __init__(self, controller, *, limits=None, user_api=None):
             limits, user_api
         )
         self._controller = controller
+        self._original_info = self._controller.info()
         self._set_threadpool_limits()
 
     def __enter__(self):
         return self
 
     def __exit__(self, type, value, traceback):
-        self.unregister()
+        self.restore_original_limits()
+
+    def restore_original_limits(self):
+        """Set the limits back to their original values"""
+        for lib_controller, original_info in zip(
+            self._controller.lib_controllers, self._original_info
+        ):
+            lib_controller.set_num_threads(original_info["num_threads"])
 
     def unregister(self):
-        for lib_controller in self._controller.lib_controllers:
-            # Since we never call get_num_threads after instanciation of
-            # ThreadpoolController, num_threads holds the original value.
-            lib_controller.set_num_threads(lib_controller.num_threads)
+        """Alias of `restore_original_limits` for backward compatibility"""
+        self.restore_original_limits()
```

Comment:
I wonder if the following wouldn't work:

```python
    # Backward compat alias
    unregister = restore_original_limits
```

Reply:
I tested it locally and it should work.

Reply:
it does indeed. thanks